crypto/asn1/a_time.c

```diff
@@ -489,7 +489,7 @@ int ASN1_TIME_print(BIO *bp, const ASN1_TIME *tm)
         if (tm->length > 15 && v[14] == '.') {
             f = &v[14];
             f_len = 1;
-            while (14 + f_len < l && isdigit(f[f_len]))
+            while (14 + f_len < l && ossl_isdigit(f[f_len]))
```

Comment:
But in this particular case, an ASN.1 UTCTime or GeneralizedTime string, the string's value is always interpreted as VisibleString/ISO646String which is really just (a subset of) ASCII, so no translation should be done here on an EBCDIC system. It should all be evaluated as if it were already ASCII (because it is). This is why I think you need a version that always treats the input as ASCII (or a proper subset of it).